src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+                    "Internal dom_eig is not supported yet!");
+    return (ARK_ILL_INPUT);
+  }
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFrequency sets dom_eig computation frequency -
+  Dominated Eigenvalue is recomputed after "nsteps" successful steps.
+
+    nsteps = 0 refers to Constant Jacobian
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (nsteps < 0)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "nsteps must be greater than or equal to 0");
+    return (ARK_ILL_INPUT);
+  }
+
+  if (nsteps == 0)
+  {
+    step_mem->const_Jac    = SUNTRUE;
+    step_mem->dom_eig_freq = 1;
+  }
+  else
+  {
+    step_mem->dom_eig_freq = nsteps;
+    step_mem->const_Jac    = SUNFALSE;
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetMaxNumStages sets the maximum number of stages allowed.
+  ---------------------------------------------------------------*/
+int LSRKStepSetMaxNumStages(void* arkode_mem, int stage_max_limit)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (stage_max_limit < 2)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "stage_max_limit must be greater than or equal to 2");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->stage_max_limit = stage_max_limit;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigSafetyFactor sets the safety factor for the DomEigs.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigSafetyFactor(void* arkode_mem, sunrealtype dom_eig_sfty)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (dom_eig_sfty < SUN_RCONST(1.0))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "dom_eig_sfty must be greater than or equal to 1");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->dom_eig_sfty = dom_eig_sfty;
```

Comment:
There should be a way to reset this to the default

Reply:
@maggul: the way that we typically do this in ARKODE is to have a clearly "illegal" value result in resetting the value to its default.  Here, you could remove the `arkProcessError` failure and instead say that if a user specified `dom_eig_sfty < 1.0`, then you just set `dom_eig_sfty` to the LSRKStep default value.